### PR TITLE
Implement the updateUsername mutation

### PR DESCRIPTION
**Pull request by samerbuna**
Includes switching the updateLanguageMutation to use the new app use-case

### CODE REVIEW

File: src/graphql/root/mutation/user-update-language.ts [hunk 1]
```diff
@@ -15,24 +15,22 @@ const UserUpdateLanguageMutation = GT.Field({
...
 
-    // TODO: redo this with app use-case
-    try {
-      user.language = language
-      await user.save()
-    } catch (err) {
-      return { errors: [{ message: err.message }] }
+    const result = wallet.updateLanguage({ language })
+
+    if (result instanceof Error) {
```
Comment (dolcalmi) [inline src/graphql/root/mutation/user-update-language.ts:27]: if you dont call updateLanguage with await then result will be always a promise

File: src/graphql/root/mutation/user-update-username.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  resolve: async (_, args, { wallet }) => {
+    const { username } = args.input
+
+    if (username instanceof Error) {
+      return { errors: [{ message: username.message }] }
+    }
+
+    const result = wallet.updateUsername({ username })
+
+    if (result instanceof Error) {
```
Comment (dolcalmi) [inline src/graphql/root/mutation/user-update-username.ts:27]: same here, result is always a promise

File: src/graphql/main/schema.graphql [hunk 3]
```diff
@@ -567,30 +568,49 @@ input AccountApiKeyDisableInput {
 
 type UserUpdateLanguagePayload {
   errors: [Error!]!
-  user: UserDetails
+  user: UserLanguageDetails
```
Comment (dolcalmi) [inline src/graphql/main/schema.graphql:571]: not sure if user is the proper name for this attribute. UserLanguageDetails only has id and language

File: src/graphql/types/payload/user-update-username.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  }),
+})
+
+const UserUpdateUsernamePayload = new GT.Object({
+  name: "UserUpdateUsernamePayload",
+  fields: () => ({
+    errors: {
+      type: GT.NonNullList(IError),
+    },
+    user: {
```
Comment (dolcalmi) [inline src/graphql/types/payload/user-update-username.ts:20]: same here, is this really user?

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: src/graphql/root/mutation/user-update-language.ts (hunk 1 above)
Comment (samerbuna) [reply src/graphql/root/mutation/user-update-language.ts:27]: Fixed

File: src/graphql/root/mutation/user-update-username.ts (hunk 2 above)
Comment (samerbuna) [reply src/graphql/root/mutation/user-update-username.ts:27]: Fixed

File: src/graphql/main/schema.graphql (hunk 3 above)
Comment (samerbuna) [reply src/graphql/main/schema.graphql:571]: It's a partial record for a user, so I think that's ok

**Review by dolcalmi — APPROVED** [review]

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**